### PR TITLE
add module goleft/indexsplit

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/goleft/indexsplit/meta.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      description: |
+        Groovy Map containing reference information
+        e.g. [ id:'genome' ]
+  - fai:
+      type: file
+      description: Reference fasta index file
+      pattern: "*.fai"
+
+output:
+  #Only when we have meta
```
Comment (nvnieuwk) [inline modules/nf-core/goleft/indexsplit/meta.yml:38]:
```suggestion
```

File: tests/modules/nf-core/goleft/indexsplit/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,8 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
+    withName: GOLEFT_INDEXSPLIT {
+        ext.args = "-n 10"
```
Comment (nvnieuwk) [inline tests/modules/nf-core/goleft/indexsplit/nextflow.config:6]: Is this a mandatory argument?

**Review by nvnieuwk — COMMENTED** [review]
I love it!

File: tests/modules/nf-core/goleft/indexsplit/nextflow.config (hunk 2 above)
Comment (matthdsm) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config:6]: yes 😈

Comment (nvnieuwk) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config:6]: Well as much as it pains me to say this but can you make this an input value? 😁

Comment (matthdsm) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config:6]: Do I have to 😞

Comment (nvnieuwk) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config:6]: I'd rather not, but the guidelines say so :/

Comment (matthdsm) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config]: fixed it

Comment (nvnieuwk) [reply tests/modules/nf-core/goleft/indexsplit/nextflow.config]: Thanks!

**Review by nvnieuwk — APPROVED** [review]

**Review by FriederikeHanssen — APPROVED** [review]

**Comment by matthdsm** [comment]
Thanks @nvnieuwk and @FriederikeHanssen

**Pull request merged**